bbconf/config_parser/bedbaseconfig.py

```diff
@@ -0,0 +1,338 @@
+        """
+        Create qdrant client object using credentials provided in config file
+
+        :return: QdrantClient
+        """
+        try:
+            return QdrantBackend(
+                collection=self._config.qdrant.collection,
+                qdrant_host=self._config.qdrant.host,
+                qdrant_port=self._config.qdrant.port,
+                qdrant_api_key=self._config.qdrant.api_key,
+            )
+        except qdrant_client.http.exceptions.ResponseHandlingException as err:
+            _LOGGER.error(f"error in Connection to qdrant! skipping... Error: {err}")
+            warnings.warn(
+                f"error in Connection to qdrant! skipping... Error: {err}", UserWarning
+            )
+
+    def _init_t2bsi_object(self) -> Union[text2bednn.Text2BEDSearchInterface, None]:
+        """
+        Create Text 2 BED search interface and return this object
+
+        :return: Text2BEDSearchInterface object
+        """
+
+        try:
+            return text2bednn.Text2BEDSearchInterface(
+                nl2vec_model=FlagEmbedding(model_name=self._config.path.text2vec),
+                vec2vec_model=self._config.path.vec2vec,
+                search_backend=self.qdrant_engine,
+            )
+        except Exception as e:
+            _LOGGER.error("Error in creating Text2BEDSearchInterface object: " + str(e))
+            warnings.warn(
+                "Error in creating Text2BEDSearchInterface object: " + str(e),
+                UserWarning,
+            )
+            return None
+
+    @staticmethod
+    def _init_pephubclient() -> Union[PEPHubClient, None]:
+        """
+        Create Pephub client object using credentials provided in config file
+
+        :return: PephubClient
+        """
+        try:
+            return PEPHubClient()
+        except Exception as e:
+            _LOGGER.error(f"Error in creating PephubClient object: {e}")
+            warnings.warn(f"Error in creating PephubClient object: {e}", UserWarning)
+            return None
+
+    def _init_boto3_client(
+        self,
+    ) -> boto3.client:
+        """
+        Create Pephub client object using credentials provided in config file
+
+        :return: PephubClient
+        """
+        try:
+            return boto3.client(
+                "s3",
+                endpoint_url=self._config.s3.endpoint_url,
+                aws_access_key_id=self._config.s3.aws_access_key_id,
+                aws_secret_access_key=self._config.s3.aws_secret_access_key,
+            )
+        except Exception as e:
+            _LOGGER.error(f"Error in creating boto3 client object: {e}")
+            warnings.warn(f"Error in creating boto3 client object: {e}", UserWarning)
+            return None
+
+    def _init_r2v_object(self) -> Region2VecExModel:
+        """
+        Create Region2VecExModel object using credentials provided in config file
+        """
+        return Region2VecExModel(self.config.path.region2vec)
```

Comment:
Not a huge concern right now, but we might need multiple region2vec models... one for each reference genome. Since they are trained to a specific reference....